File: src/main/java/com/github/spb/tget/demo/util/RandomUtils.java
package com.github.spb.tget.demo.util;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.Random;

public class RandomUtils {

    private static final String ALL_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvqxyz" +
            ",./?'|][{}()-_=+*&^:;#@$%\"0123456789";
    private static final String ALPHABETIC = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvqxyz";
    private static final String NUMERIC = "0123456789";
    private static final String SPECIAL_CHARS = ",./?'|][{}()-_=+*&^:;#@$%\"";

    private static long TICKS_AT_EPOCH = 621355968000000000L;

    private static Random random = new Random(System.currentTimeMillis() * 10000 + TICKS_AT_EPOCH);

    public static Boolean getRandomBoolean() {
        return random.nextBoolean();
    }

    public static Integer getRandomInteger() {
        return random.nextInt();
    }

    public static String generateRandomString(int charactersCount, String charactersSet) {
        List chars = Arrays.asList(charactersSet.split(""));
        StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < charactersCount; i++) {
            stringBuilder.append(getRandomElement(chars));
        }
        return stringBuilder.toString();
    }

    public static String getRandomString(int charactersCount){
        return generateRandomString(charactersCount, ALL_CHARS);
    }

    public static String getRandomAlpanumeric(int charactersCount){
        return generateRandomString(charactersCount, ALPHABETIC.concat(NUMERIC));
    }

    public static String getRandomAlphabetic(int charactersCount){
        return generateRandomString(charactersCount, ALPHABETIC);
    }

    public static String getRandomNumeric(int charactersCount){
        return generateRandomString(charactersCount, NUMERIC);
    }

    public static Object getRandomElement(Collection collection) {
        int i = random.nextInt(collection.size());
        return collection.toArray()[i];
    }
}

File: Service/src/main/java/com/github/spb/tget/demo/util/RandomUtils.kt
package com.github.spb.tget.demo.util

import java.time.LocalDateTime
import java.util.Random

class RandomUtils {

    companion object {
        private val ALL_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvqxyz" + ",./?'|][{}()-_=+*&^:;#@$%\"0123456789"
        private val ALPHABETIC = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvqxyz"
        private val NUMERIC = "0123456789"
        private val SPECIAL_CHARS = ",./?'|][{}()-_=+*&^:;#@$%\""

        private val TICKS_AT_EPOCH = 621355968000000000L

        private val random = Random(System.currentTimeMillis() * 10000 + TICKS_AT_EPOCH)

        val randomBoolean: Boolean
            get() = random.nextBoolean()

        val randomInteger: Int
            get() = random.nextInt()

        val randomEmailAddress: String
            get() = String.format("%s@%s.test",
                    getRandomAlpanumeric(15), getRandomAlpanumeric(10))

        private fun generateRandomString(charactersCount: Int, charactersSet: String = ALL_CHARS): String {
            val chars = charactersSet.split("")
            val stringBuilder = StringBuilder()
            for (i in 1 until charactersCount) {
                stringBuilder.append(getRandomElement(chars))
            }
            return stringBuilder.toString()
        }

        fun getRandomString(charactersCount: Int): String {
            return generateRandomString(charactersCount, ALL_CHARS)
        }

        fun getRandomAlpanumeric(charactersCount: Int): String {
            return generateRandomString(charactersCount, ALPHABETIC + NUMERIC)
        }

        fun getRandomAlphabetic(charactersCount: Int): String {
            return generateRandomString(charactersCount, ALPHABETIC)
        }

        fun getRandomNumeric(charactersCount: Int): String {
            return generateRandomString(charactersCount, NUMERIC)
        }

        private inline fun <reified T> getRandomElement(collection: Collection<T>): T {
            val i = random.nextInt(collection.size)
            return collection.toTypedArray()[i]
        }

        fun randomDateOfBirthAsAdult(): LocalDateTime {
            return LocalDateTime.now().minusYears(18)
        }
    }
}